modules/i18n/server/controllers/backend.server.controller.js

```diff
@@ -0,0 +1,54 @@
+const path = require('path');
+const fs = require('fs-extra');
+
+/**
+ * We want to process only one unsaved translation at the time.
+ * Otherwise the output files will be inconsistent.
+ * We make a promise chain: The next promise will start executing after the previous one finished
+ */
+const queue = { promise: Promise.resolve() };
+function enqueue(func, ...args) {
+  queue.promise = queue.promise.then(() => func(...args));
+}
+
+/**
+ * Object property order matters since ES2015
+ * So we can sort it alphabetically by keys
+ */
+const sortObject = object => Object.fromEntries(Object.entries(object).sort(([a], [b]) => a > b ? 1 : -1));
+
+/**
+ * Here we execute the standard request, not concerned with waiting
+ */
+async function processRequest(req, res) {
+  // get the needed variables
+  const { lng, ns } = req.params;
```

Comment:
`lng` is also used for "longitude" but I suspect it's the value used by default in the framework? Otherwise I'd appreciate using writing it out as `language` or `lang`. Not a big concern, just a thought.

Reply:
The [i18next documentation](https://www.i18next.com/overview/configuration-options#languages-namespaces-resources) uses `lng` and `ns`. However there is no need for us to follow that convention in our backend code. I've also replaced `ns` with more clear `namespace`.

66f2a3de28b949edfca502adcd7c941927a2a1b9